### PR TITLE
mlx-mkbfb: add -c option to check BFB

### PR DESCRIPTION
**Pull request by pgeng-nv**
RM: #4004927

### CODE REVIEW

**Comment by kblaiech** [comment]
Quick question: what is the difference between 'dump' and 'check' ?

As far as I understand, the BFB 'dump' checks the integrity of the BFB as well. Am I wrong?

**Comment by pgeng-nv** [comment]
> Quick question: what is the difference between 'dump' and 'check' ?
> 
> As far as I understand, the BFB 'dump' checks the integrity of the BFB as well. Am I wrong?

Yes you are right. “dump” also checked BFB integrity. This “check” option is implemented like a “dump” without actual dumping.

File: mlx-mkbfb [hunk 1]
```diff
@@ -391,7 +406,7 @@ class Image:
 
         self.bits = instream.read(length)
         if len(self.bits) != length:
-            raise Exception("Image ID %d: stream corrupted, image too short" %
+            raise FormatError("Image ID %d: stream corrupted, image too short" %
                             self.header.get_image_id())
```
[the remark this answers is not in the document]
Comment (pgeng-nv) [reply mlx-mkbfb:410]: No. The behavior for dump mode is exactly the same as before. Changing it to a customized exception type allow us to catch it in check mode so this exception isn't thrown.

File: mlx-mkbfb [hunk 2]
```diff
@@ -400,7 +415,7 @@ class Image:
         crc &= 0xFFFFFFFF
 
         if crc != self.header.get_image_crc():
-            raise Exception("Image ID %d: header CRC of 0x%08x does "
+            raise CRCMismatch("Image ID %d: header CRC of 0x%08x does "
```
[the remark this answers is not in the document]
Comment (pgeng-nv) [reply mlx-mkbfb:418]: Same reply as above.

**Review by lsun100 — APPROVED** [review]

**Pull request merged**